modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -117,6 +117,63 @@ export default function CartQuickAdd() {
 
 				const {name, productConfiguration, urls} = parentProduct;
 
+				if (product.discontinued) {
+					const replacementProduct = formattedProducts.find(
+						(productItem) => {
+							if (productItem.skus) {
+								return (
+									productItem.skus[0].id ===
```

Comment:
I'll have to double-check this.

But from memory, I use `skus[0]` to get the `id`, because the data that the API returns is not very well structured. That was my experience when creating this, I would test if `skus[0]` exists to determine if its a product that is not an option product (option product === `MIN1234A` or `MIN1234B`)